### PR TITLE
No delete multiple collections datasets

### PR DESCRIPTION
**Pull request by nightsh**

### CODE REVIEW

File: multi-collections.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    if isinstance(graph_file_path, (str, pathlib.Path)):
+        # load the graph from the filepath provided
+        GraphWrapper.load_graph(file_dir_path=pathlib.Path(graph_file_path).parent,
+                                file_stem_name=pathlib.Path(graph_file_path).stem)
+        # get the loaded graph object
+        graph = GraphWrapper.get_graph()
+
+        with graph.graph_lock:
+            # select datasets vertices that are in multiple collections
+            dataset_ver_seq = graph.vs.select(is_dataset_eq=True, name_ne='base_vertext').\
```
Comment (nightsh) [inline multi-collections.py:56]: This a typo? "vertext"

File: multi-collections.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    # iterate through the list of office directories and get the collection graph file
+    for office in offices_dir_list:
+        collections_graph_files.extend(office.glob('*.collections.pickle'))
+    
+    return collections_graph_files
+
+def identify_datasets_with_multi_collections(graph_file_path):
+    """ function identify datasets with multiple collections """
+
+    if isinstance(graph_file_path, (str, pathlib.Path)):
```
Comment (nightsh) [inline multi-collections.py:47]: I'm kinda missing a `else` statement here, I know it's a corner case but leads to silent fails if it ever happens. Feel free to ignore this comment, idk how likely it is to rerun this over and over again :grin:

File: multi-sources.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        # load the graph from the filepath provided
+        GraphWrapper.load_graph(file_dir_path=pathlib.Path(graph_file_path).parent,
+                                file_stem_name=pathlib.Path(graph_file_path).stem)
+        # get the loaded graph object
+        graph = GraphWrapper.get_graph()
+
+        with graph.graph_lock:
+            # select collections vertices that are in multiple sources
+            collection_ver_seq = graph.vs.select(is_collection_eq=True, name_ne='base_vertext').\
+                select(lambda vertex: 'in_source' in vertex.attribute_names() and vertex['in_source'] is not None and len(vertex['in_source']) > 1)
```
Comment (nightsh) [inline multi-sources.py:57]: :+1:

File: multi-collections.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        # load the graph from the filepath provided
+        GraphWrapper.load_graph(file_dir_path=pathlib.Path(graph_file_path).parent,
+                                file_stem_name=pathlib.Path(graph_file_path).stem)
+        # get the loaded graph object
+        graph = GraphWrapper.get_graph()
+
+        with graph.graph_lock:
+            # select datasets vertices that are in multiple collections
+            dataset_ver_seq = graph.vs.select(is_dataset_eq=True, name_ne='base_vertext').\
+                select(lambda vertex: 'in_collection' in vertex.attribute_names() and vertex['in_collection'] is not None and len(vertex['in_collection']) > 1)
```
Comment (nightsh) [inline multi-collections.py:57]: :+1:

**Comment by osahon-okungbowa** [comment]
@nightsh thanks for all this!! 👍 
will update and merge

**Pull request merged**